### PR TITLE
LPS-143802 Create GET UserGroup by ExternalReferenceCode for User Group Headless API

### PR DESCRIPTION
**Pull request by erickmonteiroo**

### CODE REVIEW

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/dto/v1_0/converter/UserGroupResourceDTOConverter.java [hunk 1]
```diff
@@ -42,6 +44,26 @@ public String getContentType() {
...
 	}
 
+	@Override
+	public UserGroup getObject(String externalReferenceCode) throws Exception {
+		UserGroup userGroup =
+			_userGroupLocalService.fetchUserGroupByExternalReferenceCode(
+				CompanyThreadLocal.getCompanyId(), externalReferenceCode);
+
+		if (userGroup == null) {
+			userGroup = _userGroupLocalService.getUserGroup(
```
Comment (drewbrokke) [inline modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/dto/v1_0/converter/UserGroupResourceDTOConverter.java:54]: Let's use `_userGroupService` for both these calls. Other converters have a weird mix so it's not consistent.

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/UserGroupResourceImpl.java [hunk 2]
```diff
@@ -59,6 +59,16 @@ public UserGroup getUserGroup(Long userGroupId) throws Exception {
...
 	}
 
+	@Override
+	public UserGroup getUserGroupByExternalReferenceCode(
+			String externalReferenceCode)
+		throws Exception {
+
+		return getUserGroup(
+			_userGroupResourceDTOConverter.getUserGroupId(
+				externalReferenceCode));
```
Comment (drewbrokke) [inline modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/UserGroupResourceImpl.java:69]: This is calling the service to get the object to get the id, then calling the service again with the id. I'd say let's just get the object directly from the DTOConverter.

**Review by drewbrokke — CHANGES_REQUESTED** [review]
@erickmonteiroo we're pretty close. Two improvements are needed. See my inline comments. Thanks!

**Review by drewbrokke — APPROVED** [review]
LGTM 👍

**Comment by drewbrokke** [comment]
ci:forward

**Comment by epark18** [comment]
Looking at these errors, the only unique failure is in _com.liferay.headless.admin.user.resource.v1_0.test.UserGroupResourceTest_ [Case Results](https://testray.liferay.com/home/-/testray/case_results/1602558331) | [Error Logs](https://testray.liferay.com/reports/production/logs/2022-01/test-1-29/test-portal-acceptance-pullrequest(master)/1424/modules-integration-mysql57-jdk8/0/1/jenkins-console.txt.gz)

`     [exec] com.liferay.headless.admin.user.resource.v1_0.test.UserGroupResourceTest > testDeleteUserGroup FAILED
     [exec]     java.lang.AssertionError: expected:<404> but was:<400>
     [exec]     	at org.junit.Assert.fail(Assert.java:89)
     [exec]     	at org.junit.Assert.failNotEquals(Assert.java:835)`

This could be a fluke though, Will test again to confirm.

**Comment by epark18** [comment]
ci:test:relevant

**Comment by erickmonteiroo** [comment]
ci:test:relevant

**Comment by erickmonteiroo** [comment]
ci:forward

**Pull request closed without merging**